components/url/lib.rs

```diff
@@ -163,6 +165,16 @@ impl ServoUrl {
 
 impl fmt::Display for ServoUrl {
     fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        //changes for bug 22485
+        if self.0.to_string().as_bytes().len() > 40 {
+            let mut hasher = DefaultHasher::new();
+            let hashed_url : String = self.0.to_string();
+            (*hashed_url).hash(&mut hasher);
+            let mut base : String = self.0.to_string();
+            base.truncate(40);
+            base.push_str(&hasher.finish().to_string());
```

Comment:
I think this would be easier to read:
```rust
let truncated: String = self.0.chars().take(40).collect()
let result = format!("{}... (...)", truncated, hasher.finish().to_string());
return result.fmt(formatter);
```

Reply:
there i think i had to do some changes, but fixed that line also